src/modules/resource/inventory.c

```diff
@@ -512,20 +551,29 @@ static int start_resource_watch (struct inventory *inv)
         flux_log_error (h, "error sending request to enclosing instance");
         goto done;
     }
-    inv->R_watch_f = f;
 
     /* Get first response synchronously
      */
     if (flux_rpc_get_unpack (f, "{s:o}", "R", &job_R) < 0) {
-        flux_log_error (h, "lookup R from enclosing instance KVS");
-        goto done;
+        if (errno == ENOSYS) {
+            /*  Parent instance doesn't support job-info.update-watch.
+             *  Fall back to job-info.lookup and return:
```

Comment:
I like the way this change doesn't disrupt the flow in the new code.

Suggestion: add a note that job-info.update-watch was added in v0.56.0.  Maybe we'll be able to drop this at some point when it seems like 0.55.0 and earlier are no longer being used.

SIde note: it would be good to have semantic versioning rules that apply to RPCs as well as APIs so we can plan deprecation without going insane once we're in v1.0 territory and beyond.